staging/src/k8s.io/sample-controller/controller.go

```diff
@@ -196,50 +198,50 @@ func (c *Controller) runWorker(ctx context.Context) {
 // processNextWorkItem will read a single work item off the workqueue and
 // attempt to process it, by calling the syncHandler.
 func (c *Controller) processNextWorkItem(ctx context.Context) bool {
-	obj, shutdown := c.workqueue.Get()
+	item, shutdown := c.workqueue.Get()
 	logger := klog.FromContext(ctx)
 
 	if shutdown {
 		return false
 	}
 
 	// We wrap this block in a func so we can defer c.workqueue.Done.
-	err := func(obj interface{}) error {
+	err := func(item interface{}) error {
 		// We call Done here so the workqueue knows we have finished
 		// processing this item. We also must remember to call Forget if we
 		// do not want this work item being re-queued. For example, we do
 		// not call Forget if a transient error occurs, instead the item is
 		// put back on the workqueue and attempted again after a back-off
 		// period.
-		defer c.workqueue.Done(obj)
-		var key string
+		defer c.workqueue.Done(item)
```

Comment:
i wonder at what point we diverged. I'm not clear on why this is paired near the Get as in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/controllers.md to ensure it's not leaked.

Reply:
Yeah, I think it would be better to do the `defer` as in controllers.md